internal/app/machined/pkg/runtime/v1alpha1/platform/metal/metal.go

```diff
@@ -94,6 +116,45 @@ func PopulateURLParameters(downloadURL string, getSystemUUID func() (string, err
 	return u.String(), nil
 }
 
+func getMacAddr() (string, error) {
+	ifen0, en0Err := net.InterfaceByName("en0")
```

Comment:
is `en0` the first interface name assigned by kernel?

Reply:
Oh, that's a great point.  I must admit, I thought of this pretty naively, just developing on my macbook, not even considering of the context it'd actually be running in on the metal.

I _believe_ the en0 to be the modern convention for the first logical ethernet device.  Great catch, I should've written a e2e test to cover this so actually ensure this works as expected.